release/pkg/filereader/file_reader.go

```diff
@@ -356,3 +362,18 @@ func ReadHttpFile(uri string) ([]byte, error) {
 
 	return data, nil
 }
+
+func ReadGitTag(projectPath, gitRootPath, branch string) (string, error) {
```

Comment:
The function name isn't representative of the behavior. Reading a git tag is a reasonably trivial operation, checking out an entire repository is a very different thing entirely: `CheckoutAndReadGitTag()` would be better and I'd suggest breaking it into 2 functions with this being a meta function.